pkg/controller/controller.go

```diff
@@ -824,11 +824,12 @@ func addOwner(meta *metav1.ObjectMeta, cr *localv1.LocalVolume) {
 	trueVal := true
 	meta.OwnerReferences = []metav1.OwnerReference{
 		{
-			APIVersion: localv1.SchemeGroupVersion.String(),
-			Kind:       localv1.LocalVolumeKind,
-			Name:       cr.Name,
-			UID:        cr.UID,
-			Controller: &trueVal,
+			APIVersion:         localv1.SchemeGroupVersion.String(),
+			Kind:               localv1.LocalVolumeKind,
+			Name:               cr.Name,
+			UID:                cr.UID,
+			Controller:         &trueVal,
+			BlockOwnerDeletion: &trueVal,
```

Comment:
Kubernetes docs says

> 	// If true, AND if the owner has the "foregroundDeletion" finalizer, then
	// the owner cannot be deleted from the key-value store until this
	// reference is removed.
	// Defaults to false.

I don't see `foregroundDeletion` set anywhere in the code.

And what is the expected outcome? The documentation above says the user cannot delete the owner (LocalVolume) without deleting owned objects (DaemonSets, ConfigMaps and whatnot), which does not make much sense.

Or is the documentation wrong?